### PR TITLE
DAOS-12501 container: Container label may not exist. (#11394)

### PR DESCRIPTION
**Pull request by wiliamhuang**
Test-tag: container test_container_user_acl datamover,-test_dm_posix_meta_entry_dcp

1. Porting 12cf6f8 to testbuild/2.3.104.
2. Added "BuildRequires: libuuid-devel" in daos.spec to avoid rpm compiling issues in CI.

daos_prop_entry_get(x, DAOS_PROP_CO_LABEL) returns NULL when the label of a container is not set.

Signed-off-by: Lei Huang <lei.huang@intel.com>

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Review by daltonbohning — APPROVED** [review]

File: utils/rpms/daos.spec
Comment (mchaarawi) [inline utils/rpms/daos.spec:113]: this change was not part of the original PR. is it carried over by accident somehow?

Comment (mchaarawi) [reply utils/rpms/daos.spec:113]: ah i see the issue. it probably is better to cherry pick the entire commit from Brian:
https://github.com/daos-stack/daos/pull/11361

Comment (wiliamhuang) [reply utils/rpms/daos.spec:105]: Thank you! Just did cherry pick [2df58d1](https://github.com/daos-stack/daos/commit/2df58d1e2d41075757b868a7c50376aaa7c49f24)

**Review by mchaarawi — APPROVED** [review]

File: utils/rpms/daos.spec
Comment (brianjmurrell) [reply utils/rpms/daos.spec:105]: Hrm.  I already have a PR in flight with the changes I have made to master that need backporting to this branch.  I guess these are going to conflict now.

Comment (wiliamhuang) [reply utils/rpms/daos.spec:105]: @brianjmurrell Thank you! Then I will wait for your PR landed then undo [2df58d1](https://github.com/daos-stack/daos/commit/2df58d1e2d41075757b868a7c50376aaa7c49f24) in this PR.

Comment (wiliamhuang) [reply utils/rpms/daos.spec:105]: @brianjmurrell Which PR were you referring to? I can refresh this one after yours is landed. Thank you!

Comment (brianjmurrell) [reply utils/rpms/daos.spec:105]: #11519

**Pull request merged**
